doc/tutorial/prng.rst

```diff
@@ -0,0 +1,720 @@
+
+>>> next_rng.name = "next_rng"
+>>> x.name = "x"
+>>> pytensor.dprint([next_rng, x], print_type=True) # doctest: +SKIP
+uniform_rv{0, (0, 0), floatX, False}.0 [id A] <RandomGeneratorType> 'next_rng'
+ |rng [id B] <RandomGeneratorType>
+ |TensorConstant{(1,) of 2} [id C] <TensorType(int64, (1,))>
+ |TensorConstant{11} [id D] <TensorType(int64, ())>
+ |TensorConstant{0.0} [id E] <TensorType(float32, ())>
+ |TensorConstant{1.0} [id F] <TensorType(float32, ())>
+uniform_rv{0, (0, 0), floatX, False}.1 [id A] <TensorType(float64, (2,))> 'x'
+<ipykernel.iostream.OutStream at 0x7f6c308c3eb0>
+
+We can see the single node with [id A], has two outputs, which we named next_rng and x. By default only the second output x is given to the user directly, and the other is "hidden".
+
+We can compile a function that returns the next_rng explicitly, so that we can use it as the input of the function in subsequent calls.
+
+>>> f = pytensor.function([rng], [next_rng, x])
+
+>>> rng_val = np.random.default_rng(123)
+>>> next_rng_val, x = f(rng_val)
+>>> print(x)
+[0.68235186 0.05382102]
+
+>>> next_rng_val, x = f(next_rng_val)
+>>> print(x)
+[0.22035987 0.18437181]
+
+>>> next_rng_val, x = f(next_rng_val)
+>>> print(x)
+[0.1759059  0.81209451]
+
+Shared variables
+================
+
+At this point we can make use of PyTensor shared variables.
+Shared variables are global variables that don't need (and can't) be passed as explicit inputs to the functions where they are used.
+
+>>> rng = pytensor.shared(np.random.default_rng(123))
+>>> next_rng, x = pt.random.uniform(rng=rng).owner.outputs
+>>> 
+>>> f = pytensor.function([], [next_rng, x])
+>>> 
+>>> next_rng_val, x = f()
+>>> print(x)
+0.6823518632481435
+
+We can update the value of shared variables across calls.
+
+>>> rng.set_value(next_rng_val)
+>>> next_rng_val, x = f()
+>>> print(x)
+0.053821018802222675
+
+>>> rng.set_value(next_rng_val)
+>>> next_rng_val, x = f()
+>>> print(x)
+0.22035987277261138
+
+The real benefit of using shared variables is that we can automate this updating via the aptly named updates kwarg of PyTensor functions.
+
+In this case it makes sense to simply replace the original value by the next_rng_val (there is not really any other operation we can do with PyTensor RNGs)
+
+>>> rng = pytensor.shared(np.random.default_rng(123))
+>>> next_rng, x = pt.random.uniform(rng=rng).owner.outputs
+>>> 
+>>> f = pytensor.function([], x, updates={rng: next_rng})
+>>> 
+>>> f(), f(), f()
+(array(0.68235186), array(0.05382102), array(0.22035987))
+
+Another way of doing that is setting a default_update in the shared RNG variable
+
+>>> rng = pytensor.shared(np.random.default_rng(123))
+>>> next_rng, x = pt.random.uniform(rng=rng).owner.outputs
+>>> 
+>>> rng.default_update = next_rng
+>>> f = pytensor.function([], x)
+>>> 
+>>> f(), f(), f()
+(array(0.68235186), array(0.05382102), array(0.22035987))
+
+Which is exactly what RandomStream does behind the scenes
```

Comment:
Emphasize a bit more the switch to `RandomStream` and link to the respective docs. We mention at the top that there are two API: numpy-like -> RandomStream, scipy-like -> pt.random.foo. We should re-emphasize that when we mention RandomStreams here.